crates/interpreter/src/instructions/memory.rs

```diff
@@ -35,3 +37,32 @@ pub fn msize(interpreter: &mut Interpreter, _host: &mut dyn Host) {
     gas!(interpreter, gas::BASE);
     push!(interpreter, U256::from(interpreter.memory.effective_len()));
 }
+// From EIP-5656 MCOPY
+pub fn mcopy<SPEC: Spec>(interpreter: &mut Interpreter, _host: &mut dyn Host) {
+    check!(interpreter, SPEC::enabled(CANCUN));           
+    pop!(interpreter, dest, src, len);
+    if len == U256::ZERO {
+        return;
+    }
+    let len = as_usize_or_fail!(interpreter, len, InstructionResult::InvalidOperandOOG);
+    gas_or_fail!(interpreter, gas::verylowcopy_cost(len as u64));
+
+
+    let dest = as_usize_or_fail!(
+        interpreter,
+        dest,
+        InstructionResult::InvalidOperandOOG
+    );
+    memory_resize!(interpreter, dest, len);
+
+
+    let src = as_usize_or_fail!(
+        interpreter,
+        src,
+        InstructionResult::InvalidOperandOOG
+    );
+    // Read data with length len from src
+    let data = interpreter.memory.clone();
```

Comment:
Cloning memory is a no go, it is expensive.

You could probably add `copy` function to memory that would done copying. Just figure out how to do in in place with vec